core/src/main/java/org/infinispan/security/actions/GetClusterExecutorAction.java

```diff
@@ -14,14 +16,27 @@
 public class GetClusterExecutorAction implements PrivilegedAction<ClusterExecutor> {
 
    private final Cache<?, ?> cache;
+   private final EmbeddedCacheManager cacheManager;
 
    public GetClusterExecutorAction(Cache<?, ?> cache) {
-      this.cache = cache;
+      this.cache = Objects.requireNonNull(cache);
+      this.cacheManager = null;
+   }
+
+   public GetClusterExecutorAction(EmbeddedCacheManager cacheManager) {
+      this.cache = null;
+      this.cacheManager = Objects.requireNonNull(cacheManager);
    }
 
    @Override
    public ClusterExecutor run() {
-      return cache.getCacheManager().executor();
+      EmbeddedCacheManager manager;
+      if (cache != null) {
+         manager = cache.getCacheManager();
```

Comment:
The cache manager methods have their own security checks, maybe we shouldn't require `ADMIN` in `SecureCacheImpl.getCacheManager()`?

Reply:
It's off-topic for this PR maybe, but I was thinking that we shouldn't need to put the `cache.getCacheManager()` call in a secure action, and we should only need a secure action for the `cacheManager.executor()` call.

Reply:
I don't feel comfortable changing that in this PR. We should probably log another JIRA to investigate that possibly.

Reply:
Ok, I created https://issues.jboss.org/browse/ISPN-10875